### ISSUE TITLE
[Auth] Azure AD B2C - Password reset

### ISSUE THREAD

**Issue by NunoBem** [microsoft/appcenter#1060]
**Describe the solution you'd like**
The user to be able to reset their password (local/email accounts).

**Describe alternatives you've considered**
We're unable to "call" the password reset policy has we do in MSAL sdk, so we had to hide the password reset label in the sign up sign in policy that Auth uses.

But having the customer forgotten the password and email us, we reset it in the B2C dashboard. This gives a temporary password so the user can sign in and give a new password. But doesn't work, just gives invalid password error. 

We don't find any other way around this, besides deleting the account and ask them to sign up again. This is an awful user experience.

**Additional context**
We implemented B2C in a former project using MSAL sdk the sign up, sign in and password reset in "email accounts". For CI/CD, analytics and crashes we used AppCenter and liked the experience. So with the new Appcenter Auth we wanted to gather everything in our new project.

.
A not so critical issue will be implementing a login with native ui [#860](https://github.com/microsoft/appcenter/issues/860), that's already an open issue.

**Comment by amchew** [comment]
Hey @nunobem good to hear from you! Thanks for the feature request, and sharing your user scenario. Looking into see if there’s a work around, will get back shortly.

**Comment by amchew** [comment]
Hey @NunoBem, hope your weekend was good! There's no workaround and we're in discussion about a fix. Will update again.

**Comment by NunoBem** [comment]
> Hey @NunoBem, hope your weekend was good! There's no workaround and we're in discussion about a fix. Will update again.

Hi @amchew, Thanks for taking time on this, it's very important to us.

**Comment by NunoBem** [comment]
@amchew  What's the feedback about this?

**Comment by amchew** [comment]
Hey @nunobem, apologies for the late reply. I just proposed a work item to create an SDK method that will accept any AAD B2C user flows or custom policies, and you'll be able to pass your "reset password" policy to App Center for your user scenario to work. The work item is currently in "ready for work" on our backlog, and should get picked up soon. Will continue to keep you updated.

**Comment by mario-merino** [comment]
@hi guys, any news regarding this?
thanks

**Comment by amchew** [comment]
Hey @mario-merino! We currently don’t have an ETA on this feature yet. If it’s a blocking feature for your apps, I’d advise you to use Azure AD B2C and the MSAL SDK directly.

**Comment by SOliasS** [comment]
@amchew Just realized that the default workflow (SignUpIn), which is the default has several errors...
The first is the ForgotPassword link which breaks login and when you click create a user and click on the back Button it also breaks...
Is there any way to get a complete working Authentication at the moment?

**Comment by madtownchris** [comment]
It's free so perhaps I shouldn't complain but what good is Auth without password reset? You say there's no workaround as yet, is that true? 

The main reason I like Auth is that it gets rid of the token broker web service for accessing a Cosmos DB. It's quite easy to implement but if users can't reset passwords it's a big problem.

**Comment by SOliasS** [comment]
@madtownchris I found a workaround detailed here #1553 .
It's using msal, but it's the only way to get a full auth flow and also .data working atm.

**Comment by SOliasS** [comment]
#1621 Auth is dead

**Comment by Zakeelm** [comment]
Thank you so much for all of your feedback. We announced this week in a [blog post]( https://devblogs.microsoft.com/appcenter/app-center-mbaas-retirement/) that we’re retiring the Auth and Data services later this year and recommending customers transition to Azure Active Directory B2C and Azure Cosmos DB instead. Let me know if you have any additional questions about this transition and i'll make sure to get them answered! Closing this issue.
